test/sdk-benchmarks/src/main/java/software/amazon/awssdk/benchmark/checksum/CheckSumBenchmark.java

```diff
@@ -0,0 +1,124 @@
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.TimeUnit;
+import software.amazon.awssdk.checksums.DefaultChecksumAlgorithm;
+import software.amazon.awssdk.checksums.SdkChecksum;
+
+@State(Scope.Benchmark)
+@Warmup(iterations = 3, time = 3, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 5, time = 5, timeUnit = TimeUnit.SECONDS)
+@Fork(1)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+public class CheckSumBenchmark {
+    @State(Scope.Thread)
+    public static class ChecksumState {
+
+        @Param( {"128B", "4KB", "128KB", "1MB"})
+        public String size;
+
+        @Param( {"MD5", "CRC32", "CRC32C", "SHA1", "SHA256"})
+        public String checksumProvider;
+
+        private byte[] payload;
+        private SdkChecksum sdkChecksum;
+
+        @Setup(Level.Trial)
+        public void setup() {
+            // Initialize the correct checksum provider based on the parameter
+            switch (checksumProvider.toUpperCase(Locale.ROOT)) {
+                case "MD5":
+                    sdkChecksum = SdkChecksum.forAlgorithm(DefaultChecksumAlgorithm.MD5);
+                    break;
+                case "CRC32":
+                    sdkChecksum = SdkChecksum.forAlgorithm(DefaultChecksumAlgorithm.CRC32);
+                    break;
+                case "CRC32C":
+                    sdkChecksum = SdkChecksum.forAlgorithm(DefaultChecksumAlgorithm.CRC32C);
+                    break;
+                case "SHA1":
+                    sdkChecksum = SdkChecksum.forAlgorithm(DefaultChecksumAlgorithm.SHA1);
+                    break;
+                case "SHA256":
+                    sdkChecksum = SdkChecksum.forAlgorithm(DefaultChecksumAlgorithm.SHA256);
+                    break;
+
+                default:
+                    throw new IllegalArgumentException("Invalid checksumProvider: " + checksumProvider);
+            }
+            // Initialize the payload based on the size parameter
+            switch (size) {
+                case "128B":
+                    payload = generateStringOfSize(128).getBytes(StandardCharsets.UTF_8);
+                    break;
+                case "4KB":
+                    payload = generateStringOfSize(4 * 1024).getBytes(StandardCharsets.UTF_8);
+                    break;
+                case "128KB":
+                    payload = generateStringOfSize(128 * 1024).getBytes(StandardCharsets.UTF_8);
+                    break;
+                case "1MB":
+                    payload = generateStringOfSize(1000 * 1024).getBytes(StandardCharsets.UTF_8);
+                    break;
+                default:
+                    throw new IllegalArgumentException("Invalid size: " + size);
+            }
+        }
+    }
+
+    @Benchmark
+    public void updateEntireByteArrayChecksum(ChecksumState state, Blackhole blackhole) {
+        state.sdkChecksum.reset();  // Ensure we reset the checksum before each run
```

Comment:
Can we do reset in tearDown methond `@TearDown`?

Reply:
Done